geniza/corpus/metadata_export.py

```diff
@@ -0,0 +1,239 @@
+    #             "textblock_set",
+    #             queryset=TextBlock.objects.select_related(
+    #                 "fragment", "fragment__collection"
+    #             ),
+    #         ),
+    #     )
+    #     .annotate(shelfmk_all=ArrayAgg("textblock__fragment__shelfmark"))
+    #     .order_by("shelfmk_all")
+    # )
+    timeprint("Done querying")
+    return docs
+
+
+def get_export_data_for_doc(doc, sep_within_cells=SEP_WITHIN_CELLS):
+    all_textblocks = doc.textblock_set.all()
+    all_fragments = [tb.fragment for tb in all_textblocks]
+    all_log_entries = doc.log_entries.all()
+    input_users = set(
+        [
+            log_entry.user
+            for log_entry in all_log_entries
+            if log_entry.user.username != script_user
+        ]
+    )
+    iiif_urls = [fr.iiif_url for fr in all_fragments]
+    view_urls = [fr.url for fr in all_fragments]
+    multifrag = [tb.multifragment for tb in all_textblocks]
+    side = [tb.side for tb in all_textblocks]
+    region = [tb.region for tb in all_textblocks]
+    old_shelfmarks = [fragment.old_shelfmarks for fragment in all_fragments]
+    libraries = set(
+        [
+            fragment.collection.lib_abbrev or fragment.collection.library
+            if fragment.collection
+            else ""
+            for fragment in all_fragments
+        ]
+    ) - {
+        ""
+    }  # exclude empty string for any fragments with no library
+    collections = set(
+        [
+            fragment.collection.abbrev or fragment.collection.name
+            if fragment.collection
+            else ""
+            for fragment in all_fragments
+        ]
+    ) - {
+        ""
+    }  # exclude empty string for any with no collection
+
+    outd = {}
+    outd["pgpid"] = doc.id
+    outd["url"] = f"{url_scheme}{site_domain}/documents/{doc.id}/"  # public site url
+    outd["iiif_urls"] = sep_within_cells.join(iiif_urls) if any(iiif_urls) else ""
+    outd["fragment_urls"] = sep_within_cells.join(view_urls) if any(view_urls) else ""
+    outd["shelfmark"] = doc.shelfmark
+    outd["multifragment"] = sep_within_cells.join([s for s in multifrag if s])
+    outd["side"] = sep_within_cells.join([s for s in side if s])  # side (recto/verso)
+    outd["region"] = sep_within_cells.join(
+        [r for r in region if r]
+    )  # text block region
+    outd["type"] = doc.doctype
+    outd["tags"] = doc.all_tags()
+    outd["description"] = doc.description
+    outd["shelfmarks_historic"] = sep_within_cells.join(
+        [os for os in old_shelfmarks if os]
+    )
+    outd["languages_primary"] = doc.all_languages()
+    outd["languages_secondary"] = doc.all_secondary_languages()
+    outd["language_note"] = doc.language_note
+    outd["doc_date_original"] = doc.doc_date_original
+    outd["doc_date_calendar"] = doc.doc_date_calendar
+    outd["doc_date_standard"] = doc.doc_date_standard
+    outd["notes"] = doc.notes
+    outd["needs_review"] = doc.needs_review
+    outd[
```

Comment:
please preserve the comment I had about this in admin (I was going to ask why you're not using `reverse` here but the comment in the old code explains why!)